src/bench/bench.cpp

```diff
@@ -8,97 +8,120 @@
 #include <assert.h>
 #include <iostream>
 #include <iomanip>
-#include <sys/time.h>
+#include <chrono>
+#include <algorithm>
+#include <regex>
 
-benchmark::BenchRunner::BenchmarkMap &benchmark::BenchRunner::benchmarks() {
-    static std::map<std::string, benchmark::BenchFunction> benchmarks_map;
-    return benchmarks_map;
+void benchmark::ConsolePrinter::header()
+{
+    std::cout << "# Benchmark, evals, iterations, total, min, max, median" << std::endl;
 }
 
-static double gettimedouble(void) {
-    struct timeval tv;
-    gettimeofday(&tv, nullptr);
-    return tv.tv_usec * 0.000001 + tv.tv_sec;
+void benchmark::ConsolePrinter::result(const State& state)
+{
+    auto results = state.m_elapsed_results;
+    std::sort(results.begin(), results.end());
+
+    double total = state.m_num_iters * std::accumulate(results.begin(), results.end(), 0.0);
+
+    size_t mid = results.size() / 2;
+    double median = results[mid];
+    if (0 == results.size() % 2) {
+        median = (results[mid] + results[mid + 1]) / 2;
+    }
+    std::cout << std::setprecision(6);
+    std::cout << state.m_name << ", " << state.m_num_evals << ", " << state.m_num_iters << ", " << total << ", " << results.front() << ", " << results.back() << ", " << median << std::endl;
 }
 
-benchmark::BenchRunner::BenchRunner(std::string name, benchmark::BenchFunction func)
+void benchmark::ConsolePrinter::footer() {}
+void benchmark::PlotlyPrinter::header()
 {
-    benchmarks().insert(std::make_pair(name, func));
+    std::cout << "<html><head>"
+              << "<script src=\"https://cdn.plot.ly/plotly-latest.min.js\"></script>"
```

Comment:
Plotly.js is released under the MIT license: https://github.com/plotly/plotly.js/blob/master/LICENSE, so that should be pretty compatible :)
The HTML output is optional, and default output is commandline, so I did not see a problem with that.
I've chosen it because the box plots are really nice, especially when using more evaluations this the output make it very visible how the runtime actually fluctuates. That makes comparisons much easier. It would be very hard to reproduce that in the command line.